### PR TITLE
Inventory for f5

### PR DESCRIPTION
**Pull request by Rishabh-Baranwal**

### CODE REVIEW

**Comment by grastogi23** [comment]
@Jeyanthinath @Rishabh-Baranwal  let us rebase it to 17.2.8

**Comment by Jeyanthinath** [comment]
@grastogi23 rebased, But please don't merge it now

File: python/avi/migrationtools/setup.py [hunk 1]
```diff
@@ -33,7 +34,7 @@
                       'pycrypto', 'ecdsa', 'pyOpenssl', 'nose-html-reporting',
                       'nose-testconfig', 'ConfigParser', 'xlsxwriter', 'jinja2',
                       'pandas', 'openpyxl', 'appdirs',  'pexpect',
-                      'unittest2', 'networkx'],
+                      'unittest2', 'networkx', 'f5-sdk', 'bigsuds'],
```
Comment (grastogi23) [inline python/avi/migrationtools/setup.py]: we cannot have these tools be installed by avi. We need to raise proper error if f5-sdk and bigsuds is not found.

**Review by grastogi23 — CHANGES_REQUESTED** [review]
let us change the dependency on f5-sdk and bigsuds

File: python/avi/migrationtools/setup.py (hunk 1 above)
Comment (Jeyanthinath) [reply python/avi/migrationtools/setup.py]: so we need to remove the dependency over here and through exceptions only at run time if the packages or not found , isn't it ?

**Comment by grastogi23** [comment]
Yes. Also this way those libraries would not get packed with controller and
customers will install them.

Gaurav Rastogi
Avi Networks


On Feb 21, 2018, at 11:58 AM, Jeyanthinath MuthuRam <
notifications@github.com> wrote:

*@Jeyanthinath* commented on this pull request.
------------------------------

In python/avi/migrationtools/setup.py
<https://github.com/avinetworks/sdk/pull/530#discussion_r169759928>:

> @@ -33,7 +34,7 @@
                       'pycrypto', 'ecdsa', 'pyOpenssl', 'nose-html-reporting',
                       'nose-testconfig', 'ConfigParser',
'xlsxwriter', 'jinja2',
                       'pandas', 'openpyxl', 'appdirs',  'pexpect',
-                      'unittest2', 'networkx'],
+                      'unittest2', 'networkx', 'f5-sdk', 'bigsuds'],

so we need to through exceptions at run time if the packages or not found ,
isn't it ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/avinetworks/sdk/pull/530#discussion_r169759928>, or mute
the thread

File: python/avi/migrationtools/f5_inventory.py [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+import os
+import argparse
+import json
+import yaml
+import xlsxwriter
+try:
+    from bigsuds import BIGIP               # version 10
+    from f5.bigip import ManagementRoot     # version 11
+except ImportError:
+    print "Please Install bigsuds and f5 sdk's to proceeed.."
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_inventory.py]: We should Log error as print may not work. Also, don't continue after this point. Simply raise the exception again.

Comment (Jeyanthinath) [reply python/avi/migrationtools/f5_inventory.py]: ok will exit the code after raising the exception

**Comment by grastogi23** [comment]
@Rishabh-Baranwal @Jeyanthinath do we still keep this out?

File: python/avi/migrationtools/setup.py (hunk 1 above)
Comment (grastogi23) [reply python/avi/migrationtools/setup.py]: this is back... we should remove 'f5-sdk', 'bigsuds' dependencies

**Review by grastogi23 — CHANGES_REQUESTED** [review]

**Comment by Jeyanthinath** [comment]
removed the package dependency from setup.py

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**